src/npcmove.cpp

```diff
@@ -1304,16 +1396,210 @@ const item_location npc::find_usable_ammo( const item &weap ) const
     return const_cast<npc *>( this )->find_usable_ammo( weap );
 }
 
+void npc::adjust_power_cbms()
+{
+    if( !is_following() || wants_to_recharge_cbm() ) {
+        return;
+    }
+    for( const bionic_id &cbm_id : power_cbms ) {
+        deactivate_bionic_by_id( cbm_id );
+    }
+}
+
+void npc::activate_combat_cbms()
+{
+    for( const bionic_id &cbm_id : defense_cbms ) {
+        activate_bionic_by_id( cbm_id );
+    }
+    if( can_use_offensive_cbm() ) {
+        for( const bionic_id &cbm_id : weapon_cbms ) {
+            check_or_use_weapon_cbm( cbm_id );
+        }
+    }
+}
+
+void npc::deactivate_combat_cbms()
+{
+    for( const bionic_id &cbm_id : defense_cbms ) {
+        deactivate_bionic_by_id( cbm_id );
+    }
+    deactivate_bionic_by_id( bio_hydraulics );
```

Comment:
It seems that hydraulics are only deactivated, but ain't activated in `activate_combat_cbms()` above.

Reply:
Hydraulics are activated when you're actually making reach or melee attacks, so as to limit unnecessary power drain when moving or if you're using ranged weapons.